src/main/java/com/cloudbees/plugins/credentials/casc/CredentialsRootConfigurator.java

```diff
@@ -77,9 +83,24 @@ public GlobalCredentialsConfiguration instance(Mapping mapping, ConfigurationCon
     @Override
     @NonNull
     public Set<Attribute<GlobalCredentialsConfiguration,?>> describe() {
-        return Collections.singleton(new Attribute<GlobalCredentialsConfiguration, SystemCredentialsProvider>("system", SystemCredentialsProvider.class)
-            .getter( t -> SystemCredentialsProvider.getInstance() )
-            .setter( noop() ));
+        HashSet<Attribute<GlobalCredentialsConfiguration, ?>> set = new HashSet<>();
+        Attribute<GlobalCredentialsConfiguration, SystemCredentialsProvider> system = new Attribute<GlobalCredentialsConfiguration, SystemCredentialsProvider>("system", SystemCredentialsProvider.class)
+                .getter(t -> SystemCredentialsProvider.getInstance())
+                .setter(noop());
+        set.add(system);
+        for(CredentialsProvider provider : CredentialsProvider.all()) {
+            Symbol s = provider.getClass().getAnnotation(Symbol.class);
+            String name;
+            if (s != null) {
+                name = s.value()[0]; // preferred
+            } else {
+                name = provider.getClass().getSimpleName(); // not so preferred
```

Comment:
should this strip off the `CredentialsProvider` suffix like jcasc does when guessing symbols normally

Reply:
Do you know of some public API I can use from CasC to get this?

Reply:
https://github.com/jenkinsci/configuration-as-code-plugin/blob/dcd748d6518204f44f7845a7afef80a8b6a601fe/plugin/src/main/java/io/jenkins/plugins/casc/impl/attributes/DescribableAttribute.java#L37

Reply:
Done. Much better now. Thanks.